### PR TITLE
Add support for nodejs and python image defaulting and upgrade

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

Defaulting and upgrading for Java auto-instrumentation already exists.

### CODE REVIEW

**Comment by pavolloffay** [comment]
@VineethReddy02 would you like to review?

**Comment by pavolloffay** [comment]
@VineethReddy02 friendly ping. Please review this one, it finishes the feature parity for all auto-instrumentations.

File: pkg/instrumentation/upgrade/upgrade.go [hunk 1]
```diff
@@ -74,5 +76,21 @@ func (u *InstrumentationUpgrade) upgrade(_ context.Context, inst v1alpha1.Instru
...
 	}
+	autoInstNodeJS := inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationNodeJS]
+	if autoInstNodeJS != "" {
+		// upgrade the image only if the image matches the annotation
+		if inst.Spec.NodeJS.Image == autoInstNodeJS {
+			inst.Spec.NodeJS.Image = u.DefaultAutoInstrNodeJS
+			inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationNodeJS] = u.DefaultAutoInstrNodeJS
+		}
+	}
+	autoInstNodePython := inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationPython]
```
Comment (VineethReddy02) [inline pkg/instrumentation/upgrade/upgrade.go:87]: `autoInstNodePython` -> `autoInstPython` node as python suffix is bit confusing.

**Review by VineethReddy02 — APPROVED** [review]
Thanks for this PR @pavolloffay 
LGTM! 

cc @jpkrohling for the final review (totally optional Juraci, if you are occupied)!

File: pkg/instrumentation/upgrade/upgrade.go (hunk 1 above)
Comment (pavolloffay) [reply pkg/instrumentation/upgrade/upgrade.go:87]: good catch this is a typo. fixed

**Comment by jpkrohling** [comment]
> cc @jpkrohling for the final review 

Your approval is enough. Merging.

**Pull request merged**